protocol/syncer.go

```diff
@@ -315,6 +326,8 @@ func (s *Syncer) BestPeer() *syncPeer {
 
 // HandleUser is a helper method that is used to handle new user connections within the Syncer
 func (s *Syncer) HandleUser(peerID peer.ID, conn *grpc.ClientConn) error {
+	s.peersLock.Lock()
+	defer s.peersLock.Unlock()
```

Comment:
There is a lag until you get the status from peer. So it might be a good idea to lock after you get response from peer.

Reply:
I agree with Yoshiki.

Please, only lock+unlock around the statements that actually use the resource.
In this case that would be around the statement on L342.

Reply:
Irrelevant after commit 9593ada